spec/takeaway_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'takeaway'
+require 'twilio'
+require 'support/shared_examples_for_twilio'
+MOCK_PRICE = 1
+
+describe Takeaway do
+
+  let(:takeaway) {described_class.new}
+  let(:dish) {double}
+  let(:class_double) {double}
+
+  context 'The menu' do
+    let(:menu_file) {double(each_line: nil)}
+
+    describe '#menu' do
+      it 'displays the list of dishes' do
+        allow(File).to receive(:read).and_return(menu_file)
+        expect(File).to receive(:read)
+        takeaway.menu
+      end
+    end
+  end
+
+  context 'Ordering dishes' do
+    describe '#take_order' do
+      it 'raises an error when user\'s cost total doesn\'t match the real total' do
+        allow(takeaway).to receive(:dish_prices).and_return(MOCK_PRICE)
+        expect{takeaway.take_order(MOCK_PRICE+1, dish)}.to raise_error 'The total provided does not match the sum of the prices of the dishes selected'
```

Comment:
Metrics/LineLength: Line is too long. [151/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)